src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,218 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Net;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Lumper.UI.ViewModels;
+using MessageBox.Avalonia.Enums;
+using MessageBox.Avalonia;
+using Avalonia.Controls;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.IO;
+
+namespace Lumper.UI.Updater
+{
+    internal class Updater
+    {
+        //struct for deserializing JSON objects
+        private struct Asset

```

Comment:
Use `/// <summary>` tags, also I'd say "For deserializing Github API response data" or something